### PR TITLE
report tested integrations and their tested versions

### PR DESCRIPTION
**Pull request by wconti27**
### What does this PR do?

This PR collects data about each integration and its specific version during testing, which is then collected by the APM Test Agent. After plugin tests run, the  Test Agent is queried for data describing each integration that was encountered during testing. This data is then stored as post-run artifacts. 

Daily, the artifacts data will be pulled by a script located in an internal DataDog repo ([here](https://github.com/DataDog/apm-shared-github-actions/blob/main/.github/scripts/get-github-actions-artifacts-js.sh)), and then uploaded to Metabase. 

Dashboards and analytics will be created/performed with the data regarding supported integrations, and will be crossed with customer dependency data.

### Motivation
Inform non-technical stakeholders of supported integrations & versions to make data-driven decisions

### Plugin Checklist

- [ ] Unit tests.
- [ ] TypeScript [definitions][1].
- [ ] TypeScript [tests][2].
- [ ] API [documentation][3].
- [ ] CircleCI [jobs/workflows][4].
- [ ] Plugin is [exported][5].

[1]: https://github.com/DataDog/dd-trace-js/blob/master/index.d.ts
[2]: https://github.com/DataDog/dd-trace-js/blob/master/docs/test.ts
[3]: https://github.com/DataDog/documentation/blob/master/content/en/tracing/trace_collection/library_config/nodejs.md
[4]: https://github.com/DataDog/dd-trace-js/blob/master/.circleci/config.yml
[5]: https://github.com/DataDog/dd-trace-js/blob/master/packages/dd-trace/src/plugins/index.js

### CODE REVIEW

**Review by rochdev — APPROVED** [review]

File: packages/dd-trace/test/plugins/agent.js [hunk 1]
```diff
@@ -37,15 +37,33 @@ function ciVisRequestHandler (request, response) {
...
+
+  // serialize the DD environment variables into a string of k=v pairs separated by comma
+  const serializedEnvVars = Array.from(ddEnvVars.entries())
+    .map(([key, value]) => `${key}=${value}`)
+    .join(',')
+
+  // add the serialized DD environment variables to the header
+  // to send with trace to the final agent destination
+  headers['X-Datadog-Trace-Env-Variables'] = serializedEnvVars
+  return headers
```
Comment (tlhunter) [inline packages/dd-trace/test/plugins/agent.js:66]: Is this return value being used?

File: packages/dd-trace/test/setup/mocha.js [hunk 2]
```diff
@@ -17,6 +17,10 @@ global.withVersions = withVersions
 global.withExports = withExports
 global.withNamingSchema = withNamingSchema
 global.withPeerService = withPeerService
+global.testAgent = {
+  pluginName: null,
+  pluginVersion: null
+}
```
Comment (tlhunter) [inline packages/dd-trace/test/setup/mocha.js:23]: Why is this using a global variable?

File: packages/dd-trace/test/plugins/agent.js (hunk 1 above)
Comment (wconti27) [reply packages/dd-trace/test/plugins/agent.js]: good point, I deleted the return

File: packages/dd-trace/test/setup/mocha.js (hunk 2 above)
Comment (wconti27) [reply packages/dd-trace/test/setup/mocha.js]: removed the use of the global

**Review by tlhunter — APPROVED** [review]

**Pull request merged**